web/web.go

```diff
@@ -0,0 +1,694 @@
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = nil
+	app.conv = conv
+	app.driver = dc.Driver
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// TODO: Add Index key statements.
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+func getSummary(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	summary := make(map[string]string)
+	for _, t := range reports {
+		var body string
+		for _, x := range t.Body {
+			body = body + x.Heading + "\n"
+			for i, l := range x.Lines {
+				body = body + fmt.Sprintf("%d) %s.\n", i+1, l) + "\n"
+			}
+		}
+		summary[t.SrcTable] = body
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(summary)
+}
+
+func getOverview(w http.ResponseWriter, r *http.Request) {
+	var buf bytes.Buffer
+	bufWriter := bufio.NewWriter(&buf)
+	internal.GenerateReport(app.driver, app.conv, bufWriter, nil, false, false)
+	bufWriter.Flush()
+	overview := buf.String()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(overview)
+}
+
+func getTypeMap(w http.ResponseWriter, r *http.Request) {
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	var typeMap map[string][]typeIssue
+	switch app.driver {
+	case "mysql", "mysqldump":
+		typeMap = mysqlTypeMap
+	case "postgres", "pg_dump":
+		typeMap = postgresTypeMap
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	// Filter typeMap so it contains just the types SrcSchema uses.
+	filteredTypeMap := make(map[string][]typeIssue)
+	for _, srcTable := range app.conv.SrcSchema {
+		for _, colDef := range srcTable.ColDefs {
+			if _, ok := filteredTypeMap[colDef.Type.Name]; ok {
+				continue
+			}
+			filteredTypeMap[colDef.Type.Name] = typeMap[colDef.Type.Name]
+		}
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(filteredTypeMap)
+}
+
+func setTypeMapGlobal(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var t map[string]string
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+
+	for k, v := range app.conv.SpSchema {
```

Comment:
I know I'm always pushing for single letter iteration variables, but maybe it might help readability to use something more descriptive, given the level of nesting of iterations going on here.

Suggest: 

```
for t, spSchema := range app.conv.SpSchema {
```

Reply:
updated as mentioned